### PR TITLE
docs: add more details on rich container

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

This pr tries to add a graph to explicitly show the architecture of rich container. And some illustration of rich container as well.

### Ⅱ. Does this pull request fix one issue?

none

### Ⅲ. Describe how you did it
none

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews
none

### CODE REVIEW

**Comment by allencloud** [comment]
Thanks a lot for your feedback. I will try to answer your question. While feel free to share more information if there is something missing in our docs. @houstar 

> I don't know that why we need --initscript here ? I think systemd is the init application that integrated in the CentOS and is enough to startup the rich container.
i.e. We only need to --rich systemd to start the rich container.

`--initscript` is used to add a prestart hook in the container which will be executed before the systemd starts. `--rich` is used to choose one type of the rich container mode from `systemd`, `dumbint`.

> From this User Case, you also specified -v /tmp:/tmp,
I have question that If there're content in the container image's /tmp and /run , how did you handle this case ?

This will handle it like the docker way. In most cases, it will override the inner container path.

> If we're running Systemd inside pouch ( so-called rich container), did we support pouch stop [container_id] has the same process as 'systemctl stop' inside pouch

I do not think rich container will do the `systemctl stop`, since in the original way, pouchd would send a terminal signal to the pid=1 process in the rich container which is systemd.

**Comment by houstar** [comment]
In reply to @allencloud ,

> --initscript` is used to add a prestart hook in the container which will be executed before the systemd starts. --rich is used to choose one type of the rich container mode from systemd, dumbint.

I'd like to --initscript is invisible to user and pre-defined the hook within /usr/libexec/pouch/hook.d directory. This pre-defined hook will be executed if the CMD is either systemd/init or dumbinit

>>From this User Case, you also specified -v /tmp:/tmp,
I have question that If there're content in the container image's /tmp and /run , how did you handle this case ?

>This will handle it like the docker way. In most cases, it will override the inner container path.

If there're contents in container images's /tmp and not specified -v /tmp:/tmp, did we'll move container images's /tmp content into temporary directory and then mount tmpfs in /tmp,  lastly move back the content from temporary dirtectoy to /tmp  ? 

> I do not think rich container will do the systemctl stop, since in the original way, pouchd would send a terminal signal to the pid=1 process in the rich container which is systemd.

There're some applications needs to be gracefully shutdown.  e.g. The Storage App.
I mentioned this since there're  argues between Systemd and Docker. 
See [Systemd vs. Docker](https://lwn.net/Articles/676831/)

**Comment by allencloud** [comment]
> There're some applications needs to be gracefully shutdown. e.g. The Storage App.
I mentioned this since there're argues between Systemd and Docker.

I have to say systemd has no this functionality to pass the SIGTERM to its sub processes. While what I wanna say is that `dumb-init` supports this. @houstar

File: docs/features/pouch_with_rich_container.md [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+Rich container is a very useful container mode when containerizing applications. This mode helps technical staff to complete packaging fat applications almost with no effort. It provides efficient ways to equip more basic software or system services except for target application in a single container. Then applications in containers could be running as smoothly as usual in VM or physical machine. This is a more generalized application-centric mode. This mode brings no invasiveness at all to both developers and operators. Especially for operators, they could have abilities to maintain applications in container with all essential tools or service processes they may need as usual.
 
 Rich container mode is not the default mode Pouch provides. It is an additional mode pouch brings to extend users' container experience. Users can still manage ordinary containers by switching rich container flag off.
 
+In a word, rich container can help enterprise to achieve the following two goals:
+
+* be compatible with legacy operation system;
```
Comment (Letty5411) [inline docs/features/pouch_with_rich_container.md]: operation system -> operating system ?

Comment (allencloud) [reply docs/features/pouch_with_rich_container.md]: Thanks. Updated. PTAL @Letty5411

**Comment by Letty5411** [comment]
LGTM

**Pull request merged**